### PR TITLE
03 bash 2

### PR DESCRIPTION
**Pull request by prosolvo4th**
fix_owner.sh

### CODE REVIEW

File: 03-Bash/fix_owner/fix_owner.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    if [ $user_id -lt 1000 ]; then
+        continue
+    fi
+    user_name=$(echo $user_line | cut -d ':' -f1)
+    user_home=$(echo $user_line | cut -d ':' -f6)
+    user_group=$(echo $user_line | cut -d ':' -f4)
+
+    # echo "$user_name=$user_id:$user_group-$user_home"
+
+    find $user_home -maxdepth 1 > homeFileList
```
Comment (yekovalyov) [inline 03-Bash/fix_owner/fix_owner.sh]: Consider using unique temporary file.

Comment (prosolvo4th) [reply 03-Bash/fix_owner/fix_owner.sh]: Should this file be stored in /tmp?

Comment (yekovalyov) [reply 03-Bash/fix_owner/fix_owner.sh]: Yes, it is usual place to temporary files.

File: 03-Bash/fix_owner/fix_owner.sh [hunk 2]
```diff
@@ -1,4 +1,27 @@
...
+    echo "$0 - check user\`s"
+    echo "-f, --force - fix without asknowledgement,"
+    echo "-h, --help - print these options."
+    exit 0
+}
+
+# check options
+if [ $# -gt 0 ]; then
+    case $1 in 
+        -h) printHelp;;
```
Comment (yekovalyov) [inline 03-Bash/fix_owner/fix_owner.sh]: You can use operator '|' for case alternatives.

File: 03-Bash/fix_owner/fix_owner.sh (hunk 1 above)
Comment (yekovalyov) [reply 03-Bash/fix_owner/fix_owner.sh]: Also mktemp utility can be used.

**Comment by yekovalyov** [comment]
Please squash all your commits into one and update your branch using force push.

File: 03-Bash/fix_owner/fix_owner.sh (hunk 2 above)
Comment (yekovalyov) [reply 03-Bash/fix_owner/fix_owner.sh]: This request was not satisfied.

File: 03-Bash/fix_owner/fix_owner.sh (hunk 1 above)
Comment (prosolvo4th) [reply 03-Bash/fix_owner/fix_owner.sh]: Used mktemp

File: 03-Bash/fix_owner/fix_owner.sh (hunk 2 above)
Comment (prosolvo4th) [reply 03-Bash/fix_owner/fix_owner.sh]: Used '|'

**Comment by yekovalyov** [comment]
This script doesn't fix group id.

File: 03-Bash/fix_owner/fix_owner.sh [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    once=0
+    for homeFile in $(cat $homeFileList)
+    do
+        if [ $once -eq 0 ]; then 
+            once=1
+            continue
+        fi
+
+        fileOwner=$(stat -c '%U' $homeFile)
+        if [ "$fileOwner" != "$user_name" ]; then
```
Comment (yekovalyov) [inline 03-Bash/fix_owner/fix_owner.sh:53]: Fixing group id is also required for this task.

Comment (prosolvo4th) [reply 03-Bash/fix_owner/fix_owner.sh:53]: Routine added

File: 03-Bash/fix_owner/fix_owner.sh [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                    echo "successfully assigned"
+                else
+                    echo "failed to assign"
+                fi
+            fi
+        fi
+
+    done
+
+done
```
Comment (yekovalyov) [inline 03-Bash/fix_owner/fix_owner.sh:100]: Final empty line is absent.

**Pull request merged**